src/main/java/com/bmc/truesight/meter/plugin/remedy/RemedyTicketsCollector.java

```diff
@@ -123,14 +125,16 @@ public void run() {
                         }//each chunk iteration
                         System.err.println("__________ [Total successful ingestion: " + totalSuccessfulIngestion + ", Total Records from Remedy :" + nMatches.longValue() + ", total iteration " + (iteration - 1) + " ]_____________");
                     }
+                } catch (RemedyReadFailedException e) {
+                    System.err.println("Reading Records from Remedy Failed, Reason :" + e.getMessage());
                 } catch (Exception e) {
-                    System.err.println("Exception occure while fetching the data" + e.getMessage());
+                    System.err.println("Exception occured while fetching the data" + e.getMessage());
                     e.printStackTrace();
                     eventSinkAPIstd.emit(Util.eventMeterTSI(Constants.REMEDY_PLUGIN_TITLE_MSG, e.getMessage(), Event.EventSeverity.ERROR.toString()));
                 } finally {
                     reader.logout(arServerContext);
                     if (isConnectionOpen) {
-                    	System.err.println("JSON RPC connection close success status : " + isConnectionOpen);
+                        System.err.println("JSON RPC connection close success status : " + isConnectionOpen);
```

Comment:
aah now it makes sense... you were printing it before closing it... and i was kinda confused...